rust/core-lib/src/editor.rs

```diff
@@ -616,6 +616,14 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.view.send_update_for_scroll(&self.text, &self.tab_ctx, first, last);
     }
 
+    /// Sets the cursor and scrolls to the beginning of the given line.
+    fn do_go_to_line(&mut self, line: u64) {
+        if let Some(line) = line.checked_sub(1) {
```

Comment:
If this is a case we expect I'd prefer to crash than to fail silently. I'd probably prefer that the frontend normalize line numbers to be zero based when making the request.